src/Bridge/NelmioApiDoc/Extractor/AnnotationsProvider/ApiPlatformProvider.php

```diff
@@ -0,0 +1,239 @@
+
+/**
+ * Creates Nelmio ApiDoc annotations for the api platform.
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ */
+class ApiPlatformProvider implements AnnotationsProviderInterface
+{
+    private $resourceNameCollectionFactory;
+    private $apiDocumentationBuilder;
+    private $resourceMetadataFactory;
+    private $filters;
+    private $operationMethodResolver;
+    private $router;
+
+    public function __construct(
+        ResourceNameCollectionFactoryInterface $resourceNameCollectionFactory,
+        ApiDocumentationBuilderInterface $apiDocumentationBuilder,
+        ResourceMetadataFactoryInterface $resourceMetadataFactory,
+        FilterCollection $filters,
+        OperationMethodResolverInterface $operationMethodResolver,
+        RouterInterface $router
+    ) {
+        $this->resourceNameCollectionFactory = $resourceNameCollectionFactory;
+        $this->apiDocumentationBuilder = $apiDocumentationBuilder;
+        $this->resourceMetadataFactory = $resourceMetadataFactory;
+        $this->filters = $filters;
+        $this->operationMethodResolver = $operationMethodResolver;
+        $this->router = $router;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getAnnotations() : array
+    {
+        $annotations = [];
+        $hydraDoc = $this->apiDocumentationBuilder->getApiDocumentation();
+        $entrypointHydraDoc = $this->getResourceHydraDoc($hydraDoc, '#Entrypoint');
+
+        foreach ($this->resourceNameCollectionFactory->create() as $resourceClass) {
+            $resourceMetadata = $this->resourceMetadataFactory->create($resourceClass);
+
+            $prefixedShortName = ($iri = $resourceMetadata->getIri()) ? $iri : '#'.$resourceMetadata->getShortName();
+            $resourceHydraDoc = $this->getResourceHydraDoc($hydraDoc, $prefixedShortName);
+
+            if ($hydraDoc) {
+                foreach ($resourceMetadata->getCollectionOperations() as $operationName => $operation) {
+                    $annotations[] = $this->getApiDoc(true, $resourceClass, $resourceMetadata, $operationName, $operation, $resourceHydraDoc, $entrypointHydraDoc);
+                }
+
+                foreach ($resourceMetadata->getItemOperations() as $operationName => $operation) {
+                    $annotations[] = $this->getApiDoc(false, $resourceClass, $resourceMetadata, $operationName, $operation, $resourceHydraDoc);
+                }
+            }
+        }
+
+        return $annotations;
+    }
+
+    /**
+     * Builds ApiDoc annotation from ApiPlatform data.
+     *
+     * @param bool             $collection
+     * @param string           $resourceClass
+     * @param ResourceMetadata $resourceMetadata
+     * @param string           $operationName
+     * @param array            $operation
+     * @param array            $resourceHydraDoc
+     * @param array            $entrypointHydraDoc
+     *
+     * @return ApiDoc
+     */
+    private function getApiDoc(
```

Comment:
Can be inlined too. PHPDoc can be removed (except the description) because it is already documented with scalar type hints.